src/IECorePython/IECoreBinding.cpp

```diff
@@ -86,7 +105,18 @@ std::string str<VEC>( VEC &x )\
 	std::stringstream s;\
 	for( unsigned i=0; i<VEC::dimensions(); i++ )\
 	{\
-		s << boost::lexical_cast<string>( x[i] );\
+		if constexpr( std::numeric_limits<VEC::BaseType>::has_infinity )\
+		{\
+			s << (\
+				x[i] == std::numeric_limits<VEC::BaseType>::infinity() ? g_positiveInfString :\
+				( x[i] == -std::numeric_limits<VEC::BaseType>::infinity() ? g_negativeInfString :\
+				boost::lexical_cast<std::string>( x[i] ) ) \
+			);\
+		}\
+		else\
+		{\
+			s << boost::lexical_cast<string>( x[i] );\
+		}\
```

Comment:
I think we should omit this special treatment for `str()`, and limit it only to `repr()`. The main requirement for `str()` is that it be simple and readable, which I think we had covered already. Only `repr()` has the requirement that it generate a string that executes in Python, so I think that's the only place it makes sense to use the Python-specific syntax.

Reply:
That makes sense to me. I reverted the `str()` treatment and added a test to verify `str()` is correct for `inf` values and squashed into https://github.com/ImageEngine/cortex/pull/1449/commits/3b46e202d0a762f96c1433fc5969fa0902c1dde8.